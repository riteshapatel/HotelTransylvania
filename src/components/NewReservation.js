/**
 * @class NewReservation component 
 * @author ritesh.patel
 * New reservation form
 */
import React, {Component} from 'react';
import {Text, View, Button, AlertIOS, Image} from 'react-native';
import CreateReservation from '../graphql/mutations/CreateReservation';
import styles from '../styles/Global.style.js';
import t from 'tcomb-form-native';
import {Mutation} from 'react-apollo';
import gql from 'graphql-tag';
const johnny = require('../assets/johnny.png');

const Form = t.form.Form;

const options = {
    auto: 'placeholders'
}

const Reservation = t.struct({
    name: t.String,
    hotelName: t.String,
    arrivalDate: t.String,
    departureDate: t.String
});

const UpdateTime = gql`
    mutation UpdateTime ($hotelName: String!) {
        updateTransactionTime(hotelName: $hotelName) @client
    }
`

class NewReservation extends Component {
    render () { 
        const hotelName = this.props.navigation.getParam('hotelName', 'NONE');

        let value = {
            hotelName: hotelName
        }
        
        return (
            <View style={styles.form_container}>
            <Image source={johnny} />
            <Mutation 
                mutation={CreateReservation}
            >
                {(createReservation, { client, data }) => (
                    <View>
                        
                        <Text style={styles.form_header}>{'\n'}New Coffin: {hotelName}</Text>
                        <Form ref={c => this._form = c} type={Reservation} value={value} options={options}/>
                        <Button type="Submit" title="Submit" onPress={(navigate) => {
                            const data = this._form.getValue();
                            if (data && data.name !== '' && data.arrivalDate !== '' && data.departureDate !== '') {
                                if (Date.parse(data.arrivalDate) > Date.parse(data.departureDate)) {
                                    AlertIOS.alert('Invalid dates')
                                } else {
                                    createReservation({variables: {
                                        data: data
                                    }})
                                    this.props.navigation.navigate('Hotels');
                                    client.mutate({
                                        mutation: UpdateTime,
                                        variables: {hotelName}
                                    });
                                }
                            } else {
                                AlertIOS.alert('Invalid values');
                            }

                        }} />                
                    </View>                    
                )}
            </Mutation>
            
            </View>
        );
    }
}

export default NewReservation;