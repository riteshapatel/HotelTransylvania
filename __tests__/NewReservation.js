/**
 * Unit test for NewReservation component 
 * @author ritesh.patel
 */
import React from 'react';
import NewReservation from '../src/components/NewReservation';
import {MockedProvider} from 'react-apollo/test-utils';

import renderer from 'react-test-renderer';

test ('snapshot NewReservation component', () => {
    const navigation = { navigate: jest.fn(), getParam: jest.fn() };
    const tree = renderer.create(<MockedProvider><NewReservation navigation={navigation} /></MockedProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});