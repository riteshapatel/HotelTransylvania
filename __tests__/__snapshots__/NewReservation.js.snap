// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot NewReservation component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/johnny.png",
      }
    }
  />
  <View>
    <Text
      style={
        Object {
          "color": "#640544",
          "fontSize": 20,
          "fontWeight": "bold",
          "paddingBottom": 10,
        }
      }
    >
      

      New Coffin: 
    </Text>
    <View
      style={Object {}}
    >
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <View
          style={Object {}}
        >
          <TextInput
            accessibilityLabel={null}
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Name"
            style={
              Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <View
          style={Object {}}
        >
          <TextInput
            accessibilityLabel={null}
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Hotel name"
            style={
              Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <View
          style={Object {}}
        >
          <TextInput
            accessibilityLabel={null}
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Arrival date"
            style={
              Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <View
          style={Object {}}
        >
          <TextInput
            accessibilityLabel={null}
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Departure date"
            style={
              Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</View>
`;
